pkg/operator/ceph/object/zone/controller.go

```diff
@@ -168,7 +168,7 @@ func (r *ReconcileObjectZone) reconcile(request reconcile.Request) (reconcile.Re
 	}
 
 	// validate the zone settings
-	err = r.validateZoneCR(cephObjectZone)
+	err = r.validateZoneCR(&cephCluster.Spec, cephObjectZone)
```

Comment:
It might easier/cleaner to just add the spec to `ReconcileObjectZone` instead of passing it and editing all functions signatures.